frontend/src/components/modal/column/ColumnOptionModal.vue

```diff
@@ -0,0 +1,216 @@
+            <small>Drag and Drop Columns to Change Order</small>
+          </v-col>
+          <v-col cols="12" sm="4">
+            <v-btn @click="addColumn"> Add Column </v-btn>
+          </v-col>
+        </v-row>
+        <v-row>
+          <v-col cols="12" sm="2">
+            <v-row v-for="i in cols.length" :key="i" class="number-rows">
+              <v-col cols="12" sm="3">
+                <h2>{{ i }}</h2>
+              </v-col>
+            </v-row>
+          </v-col>
+          <v-col cols="12" sm="10">
+            <v-card-text>
+              <draggable
+                :list="cols"
+                :animation="200"
+                ghost-class="ghost-card"
+                group="column.id"
+                id="column"
+              >
+                <ColumnCard
+                  v-for="column in cols"
+                  :key="column.id"
+                  :column="column"
+                  @deleteColumn="deleteColumn"
+                  @updateColumn="updateColumn"
+                  :tryError="invalidColumns"
+                />
+              </draggable>
+            </v-card-text>
+          </v-col>
+        </v-row>
+        <h4 class="error-message left-pad" v-if="unableToDeleteCol">
+          ** Unable to Delete Non-Empty Column **
+        </h4>
+        <h4 class="error-message left-pad" v-if="maxColumnsReached">
+          ** A Maximum of {{ MAX_COLS }} Columns Are Supported **
+        </h4>
+        <h4 class="error-message left-pad" v-if="minColumnsReached">
+          ** A Minimum of {{ MIN_COLS }} Column is Required **
+        </h4>
+        <h4 class="error-message left-pad" v-if="invalidColumns">
+          ** Ensure Each Column Has a Non-Empty Title **
+        </h4>
+        <v-card-actions>
+          <v-spacer></v-spacer>
+          <v-btn
+            color="blue-darken-1"
+            variant="text"
+            @click="this.closeClicked()"
+          >
+            Close
+          </v-btn>
+          <v-btn
+            color="blue-darken-1"
+            variant="text"
+            @click="this.saveClicked()"
+          >
+            Save
+          </v-btn>
+        </v-card-actions>
+      </v-card>
+    </v-dialog>
+  </v-row>
+</template>
+
+<script>
+import ColumnCard from './ColumnCard.vue'
+import { VueDraggableNext } from 'vue-draggable-next'
+import { ref } from 'vue'
+
+const MAX_COLS = 8
+const MIN_COLS = 1
+const cols = ref([])
+const nextColId = ref(0) // set to max of existing cols + 1
+
+export default {
+  components: {
+    ColumnCard,
+    draggable: VueDraggableNext,
+  },
+  emits: ['close', 'updateColumns'],
+  props: {
+    columns: {
+      type: Object,
+      default: undefined,
+    },
+    jobsByColumn: {
+      type: Object,
+      default: undefined,
+    },
+  },
+  data: () => ({
+    dialog: true,
+    cols,
+    unableToDeleteCol: false,
+    maxColumnsReached: false,
+    minColumnsReached: false,
+    MAX_COLS,
+    MIN_COLS,
```

Comment:
If these are constants I dont think they need to be in data.

Reply:
I get property undefined on instance when removed from data 🤔